code/src/components/Tasklist.js

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { format } from "date-fns";
+
+import tasks from "../reducers/tasks";
+
+const Tasklist = () => {
+  const taskList = useSelector((store) => store.tasks.taskitems);
+  console.log(taskList);
+
+  const dispatch = useDispatch();
+  const onTaskToggle = (TaskId) => {
+    dispatch(tasks.actions.toggleitem(TaskId));
+  };
+  const onTaskDelete = (index) => {
+    dispatch(tasks.actions.deleteitem(index));
+  };
+
+  const date = new Date();
+
+  const dateToday = format(date, "MM/dd/yyyy HH:mm");
+
+  return (
+    <section className="task-list">
+      {taskList.map((taskItem, taskIndex) => (
+        <article className="text-image" key={taskItem.id}>
+          <button
+            className="button-task"
+            onClick={() => onTaskDelete(taskItem.taskIndex)}
```

Comment:
When I click the trash icon to delete item on your project (deployed in netlify) it seems like it always deletes the first item in your state. I guess the issue can be solved if you change this line to  <button className="button-task" onClick={() => onTaskDelete(taskIndex)} . This is my guess, you need to test it! :)